python_modules/dagster/dagster_tests/core_tests/execution_tests/test_context.py

```diff
@@ -286,6 +307,84 @@ def op_annotation(context: OpExecutionContext, *args):
     op_annotation_graph.to_job(name="op_annotation_job").execute_in_process()
 
 
+def test_context_provided_to_asset_check():
+    instance = DagsterInstance.ephemeral()
+
+    def execute_assets_and_checks(assets=None, asset_checks=None, raise_on_error: bool = True):
+        defs = Definitions(assets=assets, asset_checks=asset_checks)
+        job_def = defs.get_implicit_global_asset_job_def()
+        return job_def.execute_in_process(raise_on_error=raise_on_error, instance=instance)
+
+    @asset
+    def to_check():
+        return 1
+
+    @asset_check(asset=to_check)
+    def no_annotation(context):
+        assert isinstance(context, AssetExecutionContext)
+
+    execute_assets_and_checks(assets=[to_check], asset_checks=[no_annotation])
+
+    @asset_check(asset=to_check)
+    def asset_annotation(context: AssetExecutionContext):
+        assert isinstance(context, AssetExecutionContext)
+
+    execute_assets_and_checks(assets=[to_check], asset_checks=[asset_annotation])
+
+    @asset_check(asset=to_check)
+    def op_annotation(context: OpExecutionContext):
+        assert isinstance(context, OpExecutionContext)
+        # AssetExecutionContext is an instance of OpExecutionContext, so add this additional check
+        assert not isinstance(context, AssetExecutionContext)
+
+    execute_assets_and_checks(assets=[to_check], asset_checks=[op_annotation])
+
+
+def test_context_provided_to_blocking_asset_check():
+    instance = DagsterInstance.ephemeral()
+
+    def execute_assets_and_checks(assets=None, asset_checks=None, raise_on_error: bool = True):
+        defs = Definitions(assets=assets, asset_checks=asset_checks)
+        job_def = defs.get_implicit_global_asset_job_def()
+        return job_def.execute_in_process(raise_on_error=raise_on_error, instance=instance)
+
+    @asset
+    def to_check():
+        return 1
+
+    @asset_check(asset=to_check)
+    def no_annotation(context):
+        assert isinstance(context, OpExecutionContext)
```

Comment:
@johannkm  this is the test case that shows an asset check getting an OpExecutionContext by default when it is used in `build_asset_with_blocking_check`. Note that if the user instead chose to use this asset check in the "normal" use case (ie not as a blocking check) it would get an `AssetExecutionContext` as shown in [1]